packages/sanes-chrome-extension/src/logic/persona/persona.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Algorithm,
+  ChainId,
+  publicIdentityEquals,
+  PublicIdentity,
+} from '@iov/bcp';
+import { UserProfile, WalletId } from '@iov/core';
+import { Slip10RawIndex } from '@iov/crypto';
+import { ReadonlyWallet } from '@iov/keycontrol/types/wallet';
+
+export interface AccountInfo {
+  readonly name: string;
+  readonly identities: ReadonlyMap<ChainId, PublicIdentity>;
+}
+
+export interface PersonaChainConfig {
+  readonly chainId: ChainId;
+  readonly algorithm: Algorithm;
+  readonly derivePath: (account: number) => ReadonlyArray<Slip10RawIndex>;
+}
+
+export class Persona {
+  private readonly _userProfile: UserProfile;
+  private readonly _chains: PersonaChainConfig[];
+
+  public constructor(
+    userProfile: UserProfile,
+    chains: ReadonlyArray<PersonaChainConfig>
+  ) {
+    this._userProfile = userProfile;
+    this._chains = [...chains];
+  }
+
+  public mnemonic(): string {
+    const firstWalletId = this._userProfile.wallets.value[0].id;
+    // all wallets use the same mnemonic, so we only need to check the first one
+    const mnemonic = this._userProfile.printableSecret(firstWalletId);
+    return mnemonic;
+  }
+
+  public async addChain(newChain: PersonaChainConfig): Promise<void> {
```

Comment:
This function is not used and we recreate everything when loading the persona. I suggest to remove it.

Reply:
Right, adding chains at runtime will not be needed. A new chain is always added after a config update and then a new `AccountManager` is created with different settings. I removed this now.